### ISSUE TITLE
Installation fails on AKS cluster (missing az command)

### ISSUE THREAD

**Issue by pahofmann** [keptn/keptn#490]
**Describe the bug**
Keptn installation fails on AKS cluster with the following error:
`az failed to describe the  cluster for retrieving the  property. `

**To Reproduce**
1. Run _keptn install --platform=aks_
2. Enter information
3. See error

**Expected behavior**
Installation should continue

**Environment**
Kubernetes version 1.12.8

**Additional context**
The installer fails with:

> Deploying keptn installer pod... 
> Installer pod deployed successfully. 
> Getting logs of pod installer-7494494f54-fkkqf 
> Starting installation of keptn 
> CLUSTER_IPV4_CIDR is not set, retrieve it using az. 
> az failed to describe the  cluster for retrieving the  property. 
> Error: keptn installation was unsuccessful

_az_ is installed locally and working fine. Taking a look at the installer container logs, it looks like the installer is trying to use _az_ from the container though and can't find it:

> ##########
> branch: master
> repository: https://github.com/keptn/installer
> commitlink: https://github.com/keptn/installer/commit/5fcdd6f2d826f5015d0b98815fb53b03340630c3
> repolink: https://github.com/keptn/installer/tree/5fcdd6f2d826f5015d0b98815fb53b03340630c3
> travisbuild: https://travis-ci.org/keptn/installer/jobs/551781339
> timestamp: 20190628.1528
> ##########Install on AKS
> [keptn|INFO] [2019-07-01 12:55:09] Starting installation of keptn
> [keptn|DEBUG] [2019-07-01 12:55:09] CLUSTER_IPV4_CIDR is not set, retrieve it using az.
> ./aks/installOnAKS.sh: line 44: az: command not found
> [keptn|ERROR] [2019-07-01 12:55:09] az failed to describe the  cluster for retrieving the  property.
>

**Comment by robertjahn** [comment]
Is there any addition detail in ~/.keptn/keptn-installer.log ?

**Comment by agrimmer** [comment]
Can you please locally (i.e. where the keptn CLI runs) execute the commands
- `az aks show --name YOUR_CLUSTER_NAME --resource-group YOUR_RESOURCE_GROUP --query=networkProfile.podCidr --output=tsv`
- `az aks show --name YOUR_CLUSTER_NAME --resource-group YOUR_RESOURCE_GROUP --query=networkProfile.serviceCidr --output=tsv`
and post the output?

**Comment by pahofmann** [comment]
No additional info in the keptn-installer.log, it's the same as the container log I posted above.    
The first command gives no output, the second returns 10.0.0.0/16.

**Comment by jetzlstorfer** [comment]
Thanks @pahofmann for the information. 
Did you apply any special network configuration for your cluster? 
It seems that the CIDR can not be retrieved with the commands we sent you (and these are also the commands used in our install procedure).

**Comment by pahofmann** [comment]
No special configuration. It is a fresh created cluster, I kept to the default settings when creating the network profile for it.

**Comment by jetzlstorfer** [comment]
Does your user have sufficient permissions to query the `networkProfile.podCidr`?

**Comment by pahofmann** [comment]
Yes, the user is the account owner it should have all necessary privileges.

**Comment by jetzlstorfer** [comment]
Are you using `kubenet`? According to the docs this is needed to retrieve the podCidr https://docs.microsoft.com/en-us/rest/api/aks/managedclusters/createorupdate#containerservicenetworkprofile

**Comment by robertjahn** [comment]
In my case, I too used the CLI and the network defaults.  I get values for both of those statements.

for this...  --query=networkProfile.podCidr --output=tsv .    I get ....
10.244.0.0/16

for this ....--query=networkProfile.serviceCidr --output=tsv .    I get ....
10.0.0.0/16

Keptn team -- how is this ENV value for CLUSTER_IPV4_CIDR being set?  
https://github.com/keptn/installer/blob/0.3.0/scripts/aks/installOnAKS.sh#L42

Appears that in my install, I did not get into that block of logic (line 44 of installOnAKS.sh that Patrick got ) for I dont have the 'CLUSTER_IPV4_CIDR is not set, retrieve it using az' message in my keptn install log. (see attached)
[keptn-installer.log](https://github.com/keptn/keptn/files/3349827/keptn-installer.log)

Partrick - here is how I made my cluster for reference.  Can you share how you made yours?
https://github.com/keptn-orders/keptn-orders-setup/blob/0.3.0/provisionAks.sh

**Comment by pahofmann** [comment]
I deployed the cluster via the gui. After taking a look at the template it looks like the _azure_ network plugin is used. I will check tomorrow if I can change the plugin or just redeploy the cluster.

**Comment by jetzlstorfer** [comment]
> Keptn team -- how is this ENV value for CLUSTER_IPV4_CIDR being set?
> https://github.com/keptn/installer/blob/0.3.0/scripts/aks/installOnAKS.sh#L42

This should be set by the CLI already.

**Comment by jetzlstorfer** [comment]
@pahofmann Thanks, please keep us posted.

**Comment by pahofmann** [comment]
From [this blog](https://vincentlauzon.com/2018/09/06/aks-with-kubenet-vs-azure-networking-plug-in/) i got that the difference in the network plugin used depends on wether you select the _basic_ or _advanced_ network mode during cluster deployment via GUI.

Basic will use the kubenet, advanced the azure plugin. I will recreate the cluster with the kubenet plugin and let you know if everything works fine then!

**Comment by pahofmann** [comment]
After creating the cluster with basic, and thus the kubenet plugin, the installation continued as expected:

> Deploying keptn installer pod...
> Installer pod deployed successfully.
> Getting logs of pod installer-6dbcf986b5-5b296
> Starting installation of keptn
> Testing connection to Kubernetes API
> Connection to Kubernetes API successful
> Installing Istio
> Pods in namespace istio-system are not running.
> Installing Istio failed.
> Stopping keptn installation. Already created resources are not deleted; execute the uninstallKeptn.sh script to clean-up.
> Error: keptn installation was unsuccessful

It still failed with another error. I haven't had time to look into it yet too much.  
After a quick check I realized the the recommended node size (16 cpu 64gb ram) is much bigger then what I used for this test cluster, so that could be the issue.  
Is _D16s_v3_ the smallest possible size or would a smaller one still work out for a demo environment?

**Comment by AloisReitbauer** [comment]
Looks like the Istio installation failed. As we use standard Istio this might be related to running Istio on AKS. @robertjahn  are there any known issues.

Regarding overall footprint: The team is currently working on the next keptn version with a  **significantly**  smaller hardware requirement. The heavy resource footprint is due to using Knative underneath.  We, however, cannot provide an exact ETA yet.

**Comment by pahofmann** [comment]
The istio issue was indeed caused by the node size. I deployed a bigger cluster node with 8vCPU 32GB RAM, and installation went through without issues now.

**Comment by robertjahn** [comment]
Glad to hear.   As reference...my cluster is single node 64GB/16vCPU (Standard_D16s_v3).  I observe typical usage of about 12% CPU usage and 25GB memory.  I have not seen istio install failure, but I did find over testing in keptn 022 that sometime I would get "insufficient memory" pod log messages using "kubectl describe pod" command at lower resource settings so I increased mine node size.  Keep us posted, but what your setup seems like it should be ok and I might switch to that too.
